libbeat/processors/add_cloud_metadata/add_cloud_metadata.go

```diff
@@ -358,8 +364,21 @@ func (p *addCloudMetadata) Run(event *beat.Event) (*beat.Event, error) {
 		return event, nil
 	}
 
-	// This overwrites the meta.cloud if it exists. But the cloud key should be
-	// reserved for this processor so this should happen.
+	// If cloud key exists in event already and overwrite flag is set to false, this processor will not overwrite the
+	// cloud fields. For example aws module writes cloud.instance.* to events already, with overwrite=false,
+	// add_cloud_metadata should not overwrite these fields with new values.
+	if !p.initData.overwrite {
+		for key := range event.Fields {
```

Comment:
Why do we need to do this? Are we expecting to be dealing with keys containing dots?

Reply:
@andrewkroh Yeah this is to deal with https://github.com/elastic/beats/pull/11612/files/38badaa57f14922562520ee38de7b9b5aba0aacf#diff-46124ff4b7e0a195eaff6cbeb8a90293R117 if keys containing dot 🤔 For example `"cloud.provider": "ec2"` instead of `"cloud": common.MapStr{"provider": "ec2"}`

Reply:
@ruflin `event.Fields.HasKey` only check if `cloud` field exists but it doesn't check if `cloud` field value is null or not. I'm thinking to still use `event.GetValue` and compare value with `nil`.